packages/website/ts/utils/utils.ts

```diff
@@ -287,4 +277,25 @@ export const utils = {
             window.onload = () => resolve();
         });
     },
+    getProviderType(provider: Web3.Provider): Providers | string {
+        const constructorName = provider.constructor.name;
+        let parsedProviderName = constructorName;
+        switch (constructorName) {
+            case 'MetamaskInpageProvider':
+                parsedProviderName = Providers.Metamask;
+                break;
+
+            case 'EthereumProvider':
+                parsedProviderName = Providers.Mist;
+                break;
+
+            default:
+                parsedProviderName = constructorName;
+                break;
+        }
+        if ((provider as any).isParity) {
```

Comment:
If we're moving away from constructor names, metamask has the equivalent.

```
window.web3.currentProvider.isMetaMask
true
```

Reply:
Yep, they both work. I guess potentially the `isMetaMask` flag is less likely to change then the constructor now. Will update.